### PR TITLE
feat: shared logger and silent test logs

### PR DESCRIPTION
**Pull request by antfu**
### 🔗 Linked issue

#3198 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Introduce `logger` and `useLogger` exports from `@nuxt/kit`:

```js
import { logger, useLogger } from '@nuxt/kit'
```

However consola is globally registered, this allows more semantic usage also removed the need for module authors to add another consola dependency when they want to create a namespace logger

This PR also introduces the `logLevel` option to `@nuxt/test-utils` defaulting to `1` which will be auto silent to the warning in the test environment. This option updates level of kit logger.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/schema/src/config/_common.ts [hunk 1]
```diff
@@ -739,4 +739,11 @@ export default {
   publicRuntimeConfig: {
     $resolve: (val: Record<string, any> = {}, get) => ({ ...val, app: defu(val.app, get('app')) })
   },
+
+  /**
+   * Log level of Nuxt logs.
+   *
+   * @version 3
+   */
+  logLevel: isTest ? 1 : 3
```
Comment (antfu) [inline packages/schema/src/config/_common.ts:748]: LogLevel in consola is an enum (https://github.com/unjs/consola/blob/abb016f568e8942a3c3222258757bfff14304f7c/src/logLevels.js) what do you think is the best way for users to support it Nuxt? Literals with an internal map?

Comment (pi0) [reply packages/schema/src/config/_common.ts:748]: Using numbers should be fine for now. I intend to make a ts rewrite sometime soon and improving level typijng.

Comment (pi0) [reply packages/schema/src/config/_common.ts:748]: Another idea: We could set the log level to `1` by default in consola itself when in test environment :D

Comment (pi0) [reply packages/schema/src/config/_common.ts]: Just moved `logLevel` as an option of test-utils. In the next rewrite of consola, we can directly support named levels : https://github.com/unjs/consola/issues/130

**Pull request merged**